test/test_recipes.py

```diff
@@ -7,9 +7,9 @@
 class TestRecipe(unittest.TestCase):
```

Comment:
Your tests are incomplete. There are no tests for duplicates, no tests for all CRUD of your recipes and categories